play-json/shared/src/main/scala/play/api/libs/json/JsLookup.scala

```diff
@@ -125,23 +125,14 @@ sealed trait JsLookupResult extends Any with JsReadable {
   /**
    * Tries to convert the node into a JsValue
    */
-  def toOption: Option[JsValue] = this match {
-    case JsDefined(v) => Some(v)
-    case _ => None
-  }
+  def toOption: Option[JsValue]
 
-  def toEither: Either[JsonValidationError, JsValue] = this match {
-    case JsDefined(v) => Right(v)
-    case undef: JsUndefined => Left(undef.validationError)
-  }
+  def toEither: Either[JsonValidationError, JsValue]
 
   def get: JsValue = toOption.get
 
   def getOrElse(v: => JsValue): JsValue = toOption.getOrElse(v)
-  def isEmpty: Boolean = this match {
-    case JsUndefined() => true
-    case JsDefined(_) => false
-  }
+  def isEmpty: Boolean
   def isDefined: Boolean = !isEmpty
```

Comment:
Would make MiMa unhappy (so not to be done right now) but making this one `final` would make sense for me, to ensure `isDefined` has the same behaviour for every impl